actors/builtin/multisig/multisig_state.go

```diff
@@ -33,11 +34,21 @@ func (st *State) AmountLocked(elapsedEpoch abi.ChainEpoch) abi.TokenAmount {
 	return big.Mul(unitLocked, big.Sub(big.NewInt(int64(st.UnlockDuration)), big.NewInt(int64(elapsedEpoch))))
 }
 
-func (st *State) isSigner(party address.Address) bool {
-	for _, ap := range st.Signers {
+func (st *State) isSigner(rt vmr.Runtime, party address.Address) bool {
+	for i, ap := range st.Signers {
 		if party == ap {
 			return true
 		}
+
+		if ap.Protocol() != address.ID {
+			idAddr, found := rt.ResolveAddress(ap)
+			if found {
+				st.Signers[i] = idAddr
```

Comment:
if we're going to do this, why not do the resolution in the constructor?

Reply:
The multisig actor's constructor? Because the address might not be registered yet, no?

Reply:
Ah right, I remember that now. Maybe instead of resolving addresses to ID addresses, we should resolve them all to their non-id variants for comparison and storage?

Reply:
@whyrusleeping are you explicitly changing your mind from [this statement](https://github.com/filecoin-project/specs-actors/pull/195#issuecomment-594206413)?

Note that there's no mechanism at present to resolve actor ID addresses to their reorg-stable forms; we can do it for account actors only because the pk address is stored explicitly in state.

Reply:
@anorth Yeah, thinking about it in this context, the UX of forcing those addresses to exist is pretty annoying. Resolving on check like this seems acceptable